client/odysseus/message/index.tsx

```diff
@@ -0,0 +1,74 @@
+import classnames from 'classnames';
+import { RefObject, useState } from 'react';
+import AsyncLoad from 'calypso/components/async-load';
+import type { Message } from '../types';
+
+import './style.scss';
+
+type ChatMessageProps = {
+	message: Message;
+	isLast: boolean;
+	messageEndRef: RefObject< HTMLDivElement >;
+};
+
+const ChatMessage = ( { message, isLast, messageEndRef }: ChatMessageProps ) => {
+	const [ liked, setLiked ] = useState( false );
+	const [ disliked, setDisliked ] = useState( false );
+
+	const isUser = message.role === 'user';
+
+	const onLike = () => {
+		if ( liked ) {
+			setLiked( false );
+			setDisliked( false );
+			return;
+		}
+		setLiked( true );
+		setDisliked( false );
+	};
+
+	const onDislike = () => {
+		if ( disliked ) {
+			setLiked( false );
+			setDisliked( false );
+			return;
+		}
+		setLiked( false );
+		setDisliked( true );
+	};
+
+	return (
+		<div
+			ref={ isLast ? messageEndRef : null }
+			className={ `chatbox-message ${ isUser ? 'user' : 'wapuu' }` }
```

Comment:
A `user` or `wapuu` classnames are a bit generic and could potentially be used somewhere else, leading to unexpected styles here. Should we prefix them?